src/lib/routers/history.ts

```diff
@@ -26,6 +26,13 @@ type BrowserHistoryProps = {
   parseURL: ParseURL;
 };
 
+type BrowserHistoryArgs = {
+  windowTitle?: (routeState: RouteState) => string;
+  writeDelay?: number;
+  createURL?: CreateURL;
+  parseURL?: ParseURL;
+};
+
```

Comment:
Just above this you have `BrowserHistoryProps` with the almost same shape.
Since it is only used to define the type of the private properties on `BrowserHistory`, you could use `BrowserHistoryArgs` for that and completely get ride of `BrowserHistoryProps`.

You'll just have to mark some properties as required, ex:
```js
  /**
   * Parses the URL into a route state.
   * It should be symetrical to `createURL`.
   */
private readonly parseURL: Required<BrowserHistoryArgs>['parseURL'];
```